mne_bids_pipeline/steps/source/_04_make_forward.py

```diff
@@ -102,7 +102,19 @@ def get_input_fnames_forward(*, cfg, subject, session):
         check=False,
     )
     in_files = dict()
-    in_files["info"] = bids_path.copy().update(**cfg.source_info_path_update)
+    # for consistency with 05_make_inverse, read the info from the
+    # data used for the noise_cov
+    if cfg.source_info_path_update is None:
+        if cfg.noise_cov in ("rest", "noise"):
+            source_info_path_update = dict(
+                processing="clean", suffix="raw", task=cfg.noise_cov
+            )
+        else:
+            source_info_path_update = dict(suffix="ave")
+            # XXX is this the right solution also for noise_cov = 'ad-hoc'?
```

Comment:
Yes it should be the best we can do -- for ad-hoc we just get the rank based on the number of projectors + maxwell info.
```suggestion
            source_info_path_update = dict(suffix="ave")
```